src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -3660,71 +3663,75 @@ int QgsOracleProviderMetadata::listStyles( const QString &uri,
     QStringList &descriptions,
     QString &errCause )
 {
+  int res = -1;
   QgsDataSourceUri dsUri( uri );
 
   QgsOracleConn *conn = QgsOracleConn::connectDb( dsUri, false );
   if ( !conn )
   {
     errCause = QObject::tr( "Could not connect to database" );
-    return -1;
+    return res;
```

Comment:
Wouldn't be better to return explicitly -1 on error so when you read the code you don't have to check if the res value has been updated somewhere ?